// _@antv_f2@3.6.0-alpha.2@@antv/f2/lib/chart/controller/coord.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports["default"] = void 0;

var _index = _interopRequireDefault(require("../../coord/index"));

var _common = require("../../util/common");

var CoordController =
/*#__PURE__*/
function () {
  function CoordController(_ref) {
    var plot = _ref.plot;
    // 要记录下plot
    this.plot = plot;
    this.cfg = {
      plot: plot
    };
  }

  var _proto = CoordController.prototype;

  _proto.create = function create(cfg) {
    // 默认直角坐标系
    var _cfg$type = cfg.type,
        type = _cfg$type === void 0 ? 'cartesian' : _cfg$type,
        plot = cfg.plot;

    var C = _index["default"][(0, _common.upperFirst)(type)];

    var coord = new C(cfg);
    this.plot = plot;
    this.cfg = cfg;
    this.coord = coord;
    return coord;
  };

  _proto.render = function render(type, config) {
    var cfg = this.cfg,
        coord = this.coord;

    if (!type && !config) {
      if (coord) {
        return coord;
      }

      return this.create(cfg);
    } // 有新的参数变化


    var coordCfg;

    if ((0, _common.isObject)(type)) {
      coordCfg = type;
    } else {
      coordCfg = config || {};
      coordCfg.type = type;
    }

    var newCfg = (0, _common.mix)({
      plot: this.plot
    }, coordCfg); // 如果配置有变化，再重新创建实例

    if (!(0, _common.isEqual)(cfg, newCfg)) {
      return this.create(newCfg);
    }

    return this.coord;
  };

  return CoordController;
}();

var _default = CoordController;
exports["default"] = _default;
